### PR TITLE
#161966897 CRUD tests for article 

### PR DESCRIPTION
**Pull request by kamyaD**
#### What does this PR do?

This PR contains failing CRUD tests for articles

#### Description of Task to be completed?

- add a test for creating an article
- add a test for retrieving  an article
- add test for editing an article
- add a test for  deleting an article

#### How should this be manually tested?

- Clone the repo and cd into the project folder
- Create a virtual environment by running `$ python3 -m venv <name_of_environment>`
- Activate the virtual environment by running `source <name_of_environment>/bin/activate`
- Run `$ pip install -r requirements.txt`
- run python manage.py test

#### What are the relevant pivotal tracker stories?

[ #161966897 ](https://www.pivotaltracker.com/story/show/161966897)
#### Screenshots (if appropriate)
image

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (r-wambui) [inline .vscode/settings.json]: Please remove the vscode. Why do we still have this I had talked about it in the last PR

File: authors/settings.py [hunk 2]
```diff
@@ -138,6 +138,6 @@
     'NON_FIELD_ERRORS_KEY': 'error',
 
     'DEFAULT_AUTHENTICATION_CLASSES': (
-        'authors.apps.authentication.backends.JWTAuthentication',
+        # 'authors.apps.authentication.backends.JWTAuthentication',
```
Comment (r-wambui) [inline authors/settings.py:151]: Do not push commented code.

File: authors/urls.py [hunk 3]
```diff
@@ -19,6 +19,6 @@
 urlpatterns = [
     url(r'^admin/', admin.site.urls),
 
-    url(r'^api/', include('authors.apps.authentication.urls', namespace='authentication')),
+    url(r'^api/', include('authors.apps.authentication.urls')),
```
Comment (r-wambui) [inline authors/urls.py]: We should have this reflect implementation of urls in django 2.

File: authors/apps/tests/test_article.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
+from django.test import TestCase
+# from ..models import Article
```
Comment (r-wambui) [inline authors/apps/tests/test_article.py]: Since you created the articles model why do we still have this?

Comment (kamyaD) [reply authors/apps/tests/test_article.py]: Hello Rose, I am still cooking something, my next push will not have these errors. Please bear with me.

**Comment by codeplus254** [comment]
Great work! Remember to squash the commits after you finish cooking :-P so that the review process is easy.

**Comment by PromasterGuru** [comment]
@kamyaD resolve the conflicts as you work on the other feedbacks.

**Comment by kamyaD** [comment]
@r-wambui I have just pushed my latest changes which has all the feedbacks that you mentioned. Please have a look and let me know. Thanks

**Comment by kamyaD** [comment]
@PromasterGuru  I have resolved the conflicts as you mentioned. Thanks for the ask feedback

File: authors/apps/articles/tests/test_views.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+            reverse('comment'), self.comment_data, format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_user_can_get_all_comments(self):
+        response = self.client.get(reverse('all_comments', ))
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    # def test_user_can_update_comments(self):
```
Comment (amutava) [inline authors/apps/articles/tests/test_views.py]: why do you have commented code on your PR?

**Comment by r-wambui** [comment]
@kamyaD Please rebase develop. I merged your other PR. Also resolve the merge conflicts

File: authors/apps/articles/tests/test_views.py [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    #     self.client.post(
+    #         reverse('update_comment'), self.comment_data, format="json")
+    #     comment = Comment.objects.get()
+    #     self.change_article = {'title': 'The love storry'}
+    #     res = self.client.put(
```
Comment (amutava) [inline authors/apps/articles/tests/test_views.py]: remove commented code

**Comment by leonardnjura** [comment]
Good work @kamyaD , Though, let's strive for utter dryness so code climate won't sense so much global warming and complain that our Infinite selves are becoming an endangered species

**Comment by kamyaD** [comment]
@r-wambui @andela-amutava I have worked on the feedbacks and rebased tests @PromasterGuru branches to have his tests too so that they could be reflected in this PR.

File: authors/apps/articles/tests/test_views.py (hunk 5 above)
Comment (kamyaD) [reply authors/apps/articles/tests/test_views.py]: Worked on this.

**Pull request merged**